### PR TITLE
Use namespace

### PR DESCRIPTION
**Pull request by gannan08**

### CODE REVIEW

File: lib/main.js [hunk 1]
```diff
@@ -11,63 +11,138 @@ import './config.js';
...
+    logger.debug('Application identity not created.', {identity});
+  }
+
+  if(!seeds.services) {
+    logger.debug('Service identity seeds not found.');
+    return;
+  }
+
+  for(const name of Object.keys(seeds.services)) {
+    const seedDescription = seeds.services[name];
```
Comment (dlongley) [inline lib/main.js]:
```suggestion
  for(const [name, seedDescription] of Object.entries(seeds.services)) {
```

File: lib/main.js [hunk 2]
```diff
@@ -11,63 +11,138 @@ import './config.js';
...
+    // any error thrown here should not be caught, we want startup to fail
+    await _generateServiceIdentity({name, seedDescription});
+    const {id} = seeds.app;
+    logger.debug('Service identity created.', {key: {name, id}});
```
Comment (dlongley) [inline lib/main.js]: I think `{key: {name, id}}` will be confusing. If you want to include the app information, then maybe: `{app: {name: 'app', id}, identity: {name, id: seedDescription.id}}` or something like that.

File: lib/main.js [hunk 3]
```diff
@@ -11,63 +11,138 @@ import './config.js';
...
+      identityId: seedDescription.id,
+      generatedKeyId: id
+    });
+  }
+
+  if(APPLICATION_IDENTITY) {
+    throw new BedrockError(
+      `Application identity was previously configured.`,
+      'DuplicateError');
+  }
```
Comment (dlongley) [inline lib/main.js]: Seems like this could go at the top.

File: lib/main.js [hunk 4]
```diff
@@ -11,63 +11,138 @@ import './config.js';
...
+});
+
+async function _generateApplicationIdentity({name, seedDescription}) {
+  const {seedBase58} = seedDescription;
+  const {id, keys} = await _generateKeys({seedBase58});
+
+  if(id !== seedDescription.id) {
+    const msg = 'The generated application id does not match the id found ' +
+      `in the identity's description.`;
```
Comment (dlongley) [inline lib/main.js]:
```suggestion
    const msg = 'The generated application ID does not match the ID found ' +
      `in the identity's description.`;
```

File: lib/main.js [hunk 5]
```diff
@@ -11,63 +11,138 @@ import './config.js';
...
+    throw new BedrockError(msg, 'InvalidStateError', {
+      identityName: name,
+      identityId: seedDescription.id,
+      generatedKeyId: id
```
Comment (dlongley) [inline lib/main.js]:
```suggestion
      expectedId: seedDescription.id,
      actualId: id
```

File: lib/main.js [hunk 6]
```diff
@@ -11,63 +11,138 @@ import './config.js';
...
+      seedName: name,
+      seedDescriptionId: seedDescription.id,
+      generatedKeyId: id
+    });
+  }
+
+  if(SERVICE_IDENTITIES.has(name)) {
     throw new BedrockError(
-      'Application key seed is not set.', 'InvalidStateError');
+      `Service identity [${name}] was previously configured.`,
```
Comment (dlongley) [inline lib/main.js]: We use parens in other places:

```suggestion
      `Service identity (${name}) was previously configured.`,
```

File: lib/main.js [hunk 7]
```diff
@@ -11,63 +11,138 @@ import './config.js';
...
+    const msg = 'The generated service id does not match the id found ' +
+      `in the service identity's description.`;
+    throw new BedrockError(msg, 'InvalidStateError', {
+      seedName: name,
+      seedDescriptionId: seedDescription.id,
+      generatedKeyId: id
+    });
+  }
+
+  if(SERVICE_IDENTITIES.has(name)) {
```
Comment (dlongley) [inline lib/main.js]: Again, seems like it could be moved to the top.

**Review by dlongley — APPROVED** [review]
Approving conditionally -- see comments. Thanks!

File: util/generate-seed.js [hunk 8]
```diff
@@ -3,17 +3,46 @@
...
-  console.log(`SECRET seedBase58: ${seedBase58}`);
+  // generate a random base58 encoded 256 Bit (32 Byte) seed
+  const seedBase58 = await generateId({
+    bitLength: SEED_BITS_SIZE, multibase: false
+  });
   // convert base58 to Uint8Array
-  const seed = decode(seedBase58);
-  const didKey = await didKeyDriver.generate({seed});
+  const seedBytes = decode(seedBase58);
+  if(seedBytes.length !== SEED_BYTE_SIZE) {
```
Comment (dlongley) [inline util/generate-seed.js:28]: Why is this check necessary? We trust `generateId` or we don't, right?

Comment (dlongley) [reply util/generate-seed.js:28]: Can't we just use bnid's [IdGenerator](https://github.com/digitalbazaar/bnid#reusable-components) and avoid having to decode here?

Comment (gannan08) [reply util/generate-seed.js:28]: We can use it. Pushing an update.

**Pull request merged**